nodejs/eks/cluster.ts

```diff
@@ -109,14 +109,19 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     }
 
     // Create the EKS service role
-    const eksRole = new ServiceRole(`${name}-eksRole`, {
+    let eksRole: pulumi.Output<aws.iam.Role>;
+    if (args.serviceRole) {
+      eksRole = pulumi.output(args.serviceRole);
+    } else {
+      eksRole = (new ServiceRole(`${name}-eksRole`, {
         service: "eks.amazonaws.com",
         description: "Allows EKS to manage clusters on your behalf.",
         managedPolicyArns: [
             "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
             "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
         ],
-    }, { parent: parent });
+      }, { parent: parent })).role;
```

Comment:
I appreciate the changes for brevity in accessing and only using the `.role` here and below in the lambda for getting its ARN, but I wonder if for future cases there may be a need to export the details of the entire `ServiceRole` versus just extracting the role here and tossing out the rest of the object props.

Reply:
Thank you for all the feedback, I've accepted your other suggestions. 

Regarding this one: the reason for doing this was because the changes would pervade deeper if I were to create a ServiceRole in both logical paths (role is injected / role is created). We would need to modify the servicerole class to allow the role to be injected. I can proceed along this route if you think this is preferable.